### DISCUSSION TITLE
How get the details (or RAW info) of an article with enclosure ?

### DISCUSSION THREAD

**Discussion by Thomas-AA** [martinrotter/rssguard#1342] [Q&A] [answered]
Hi,

is it possible in the `nowebengine` version. To get the urls of the `enclosed` documents ? or at the very least see the raw data of an article ?

Thanks.

**Comment by martinrotter** [comment]
You mean to display HTML/source of the article? Or the source XML of the whole feed?

**Reply by Thomas-AA** [reply]
Hi @martinrotter , thanks for your extra fast reaction 😱 

Yes as workaround if RSSguard `nowebengine` do not offer yet a way to retrieve the url's of the enclosed content (for the article)
is it possible to see the raw code of the article ? I believe it's XML then. in order to retrieve those url's ? thanks.

**Reply by martinrotter** [reply, marked as answer]
It is possible by abusing article filtering feature (Article Filters dialog).

```js
function filterMessage() {
  console.log(msg.rawContents);

  return MessageObject.Accept;
}
```

Assign the filter to all feeds you are interested in. Then if you run "Test" button, it will print all raw contents of all articles from checked/selected feeds into command line output.

Well, this is highly advanced stuff. I guess I could add the feature. It is true, that WebEngine version has more feature in this aspect than lite one. I will maybe implement it the way so both web backends share exactly same features in this aspect to avoid user confusion.

**Reply by Thomas-AA** [reply]
@martinrotter , 

I've copy/pasted your code and tick a feed.

When I push `test` it highlight the articles of this feed in the `Existing articles` tab in green.
in the `script output` it's only related to the `sample article` tab..

I see in your script
```JS
console.log();
```
So I suppose the output is not intended to be in the `script output` tab but rather in the RSSguard console. But how display it ? I've searched `console` in https://rssguard.readthedocs.io/en/stable/search.html?q=console&check_keywords=yes&area=default but this didn't help me.

Thanks.

**Reply by martinrotter** [reply]
Well You can display console:

Main menu -> About > Show application log

It is non-modal dialog, so open the log first, then go to article filters window and do your thing

**Reply by martinrotter** [reply]
Here is sample screenshot of how it looks when article raw contents is printed to log.

![image](https://github.com/martinrotter/rssguard/assets/1255302/e69859f3-1b96-4c1f-81cc-f22dcd1d5ee0)

**Reply by Thomas-AA** [reply]
Thanks again @martinrotter I've found the console (shame on me, it's was so easy: help ->  Display application log.

So your trick work. but it certainly not the best to work with. ( of course it's a workaround )

Cheers

**Reply by martinrotter** [reply]
Yes, the feature was not really meant for this use-case.
